components/net/fetch/methods.rs

```diff
@@ -264,10 +264,18 @@ pub fn main_fetch(request: Rc<Request>,
         }
 
         // Step 16
-        // TODO this step (CSP/blocking)
+        // TODO Blocking for CSP, mixed content, MIME type
+        let blocked_error_response;
+        let internal_response = if should_block_nosniff(&request, &response) {
+            // Defer rebinding result
+            blocked_error_response = Response::network_error(NetworkError::Internal("Blocked by nosniff".into()));
+            &blocked_error_response
+        } else {
+            internal_response
+        };
 
         // Step 17
-        if !response.is_network_error() && (is_null_body_status(&internal_response.status) ||
+        if !internal_response.is_network_error() && (is_null_body_status(&internal_response.status) ||
```

Comment:
This is a mind twister, and it either requires documentation, or a [bug filed against the spec](https://github.com/whatwg/fetch), or both. Let's say both.